src/results/aggregator.rs

```diff
@@ -85,9 +85,13 @@ pub async fn aggregate(
             .https_only(true)
             .gzip(true)
             .brotli(true)
-            .http2_adaptive_window(config.adaptive_window)
-            .build()
-            .unwrap()
+            .http2_adaptive_window(config.adaptive_window);
+
+        if config.proxy.is_some() {
+            cb = cb.proxy(config.proxy.clone().unwrap());
+        }
+
+        cb.build().unwrap()
```

Comment:
```suggestion
            .http2_adaptive_window(config.adaptive_window);
        
        #[cfg(feature = "socks")]
        if config.proxy.is_some() {
            cb = cb.proxy(config.proxy.clone().unwrap());
        }

        cb.build().unwrap()
```

I would suggest having this behind a `cfg` feature macro as this will only add the following code to the binary if the `socks` feature has been enabled during build otherwise it will not include it thus reducing binary size and removing unneeded code. 🙂